internal/backoff/exponential.go

```diff
@@ -35,8 +35,7 @@ type ExponentialOption func(*exponentialOptions)
 // exponentialOptions are the configuration options for an exponential backoff
 type exponentialOptions struct {
 	base, min, max time.Duration
```

Comment:
Also...consider whether we should choose a new name for `base`. (Perhaps `initial`?)

In the context of exponentiation, I expected `base` to be `M` in `M ^ n`. That made me immediately suspicious of the `time.Duration` type.